src/renderers/dom/client/wrappers/ReactDOMSelect.js

```diff
@@ -58,6 +58,28 @@ function warnIfValueIsNull(props) {
   }
 }
 
+function warnIfDuplicateValues(inst) {
+  const options = inst._currentElement.props.children;
+
+  if (options === undefined) {
+    return;
+  }
+
+  // Displays a warning for all duplicate values in select element. Does not exit after first duplicate is found.
+  for (var i = 0; i < options.length-1; i++) {
+    if (options[i].props.value !== undefined) {
```

Comment:
Children are not required to be only elements. This needs to check the type (also related to composite components as mentioned by @jimfb, shouldn't check the value of those).